metals/src/main/scala/scala/meta/internal/worksheets/WorksheetProvider.scala

```diff
@@ -274,22 +319,30 @@ class WorksheetProvider(
     val mdoc = getMdoc(path)
     val input = path.toInputFromBuffers(buffers)
     val relativePath = path.toRelative(workspace)
-    val worksheet = mdoc.evaluateWorksheet(relativePath.toString(), input.value)
-    val classpath = worksheet.classpath().asScala.toList
+    val evaluatedWorksheet =
+      mdoc.evaluateWorksheet(relativePath.toString(), input.value)
+    val classpath = evaluatedWorksheet.classpath().asScala.toList
     val previousDigest = worksheetsDigests.getOrElse(path, "")
     val newDigest = calculateDigest(classpath)
 
+    exportableEvaluations.update(
```

Comment:
Should we remove a key if a file was closed?

Reply:
I _don't think_ it really matters. Mainly because if you close a worksheet, you won't be able to send in a request to get the evaluations for that URI until it's open again, and that opening would re-evaluate and update this anyways. So I don't think there is much there we need to worry about.